### PR TITLE
Fetch the house and senators list and save it to the DB

### PR DESCRIPTION
**Pull request by thearavind**

### CODE REVIEW

File: octopus.go [hunk 1]
```diff
@@ -1,49 +1,36 @@
...
-	logger.Log("Fetch number of bytes from:", url, "Data:", len(data))
-
-	var resp map[string]interface{}
-	err = json.Unmarshal(data, &resp)
+	senators := models.MemberListResposne{}
+	err := workers.Fetch(115, workers.Senate, workers.Member, ".json", &senators)
 	if err != nil {
-		logger.Log("Failed to decode json with error:", err)
-		return
+		logger.Log("Failed to fetch senators data", "Error: ", err)
```
Comment (ravi-lattis) [inline octopus.go]: I guess, you missed "return err" here.

File: octopus.go [hunk 2]
```diff
@@ -1,49 +1,36 @@
...
+	logger.Log("Saved the senators data to the DB")
 
-	logger.Log("Payload:", payload)
-	payloadJson, err := json.Marshal(payload)
+	house := models.MemberListResposne{}
+	err = workers.Fetch(115, workers.House, workers.Member, ".json", &house)
 	if err != nil {
-		logger.Log("Could not encode payload as json:", err)
-		return
+		logger.Log("Failed to fetch house members data", "Error: ", err)
```
Comment (ravi-lattis) [inline octopus.go]: "return err" here too.

File: octopus.go [hunk 3]
```diff
@@ -1,49 +1,36 @@
...
 	}
+	logger.Log("Fetched the house members data", house, ",saving it to the DB")
 
-	var sens []models.Senator
-	err = json.Unmarshal(payloadJson, &sens)
-	if err != nil {
-		logger.Log("Could not decode senators", err)
-		return
+	houseLength := house.Results[0].NumResults
+	for i := 0; i < houseLength; i++ {
```
Comment (ravi-lattis) [inline octopus.go]: Store the length in a variable and use to make the loop more faster.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+*.log
+.idea
```
Comment (greenac) [inline .gitignore]: This should not be but in the project .gitignore, but in your global git ignore. I'm not sure where that lives on a windows machine (On a mac it is `/Users/<username>/.gitignore_global`). Also there should be a new-line at the end of this file.

File: config/config.go [hunk 5]
```diff
@@ -1,47 +1,56 @@
...
+	Password              string
+	MembersCollection     string
+	SenatorsCollection    string
 	LegislativeCollection string
 }
 
+//LogInfo - Config strings related to the logger
 type LogInfo struct {
-	Path string
+	Path  string
```
Comment (greenac) [inline config/config.go]: Don't think we need the extra space here.

File: config/config.go [hunk 6]
```diff
@@ -1,47 +1,56 @@
...
 
+//ApiKeys - list of all the api keys
 type ApiKeys struct {
 	ProPublicaCongress string
 }
 
+//Config - Octopus config
 type Config struct {
-	Mongo Mongo
+	Mongo   Mongo
```
Comment (greenac) [inline config/config.go]: extra space

**Review by greenac — COMMENTED** [review]
hey @thearavind can you take a look at these comments?

File: models/models.go [hunk 7]
```diff
@@ -1,5 +1,52 @@
 package models
 
-type Model interface {
-	toMap() map[string]interface{}
+//MemberListResposne - Memeber response format
+type MemberListResposne struct {
```
Comment (ravirajsubramanian) [inline models/models.go]: `MemberListResposne` Spelling mistake.

File: octopus.go [hunk 8]
```diff
@@ -1,49 +1,36 @@
...
-	var resp map[string]interface{}
-	err = json.Unmarshal(data, &resp)
+	senators := models.MemberListResposne{}
+	err := workers.Fetch(115, workers.Senate, workers.Member, ".json", &senators)
 	if err != nil {
-		logger.Log("Failed to decode json with error:", err)
-		return
+		logger.Log("Failed to fetch senators data", "Error: ", err)
 	}
+	logger.Log("Fetched the senators data", senators, ",saving it to the DB")
```
Comment (ravirajsubramanian) [inline octopus.go]: `else` condition missing.

File: octopus.go [hunk 9]
```diff
@@ -1,49 +1,36 @@
...
-	logger.Log("Payload:", payload)
-	payloadJson, err := json.Marshal(payload)
+	house := models.MemberListResposne{}
+	err = workers.Fetch(115, workers.House, workers.Member, ".json", &house)
 	if err != nil {
-		logger.Log("Could not encode payload as json:", err)
-		return
+		logger.Log("Failed to fetch house members data", "Error: ", err)
 	}
+	logger.Log("Fetched the house members data", house, ",saving it to the DB")
```
Comment (ravirajsubramanian) [inline octopus.go]: `else` condition again.

File: octopus.go [hunk 10]
```diff
@@ -1,49 +1,36 @@
...
 
-	var sens []models.Senator
-	err = json.Unmarshal(payloadJson, &sens)
-	if err != nil {
-		logger.Log("Could not decode senators", err)
-		return
+	houseLength := house.Results[0].NumResults
+	for i := 0; i < houseLength; i++ {
+		db.Connection.MemberCollection.Insert(&house.Results[0].Members[i])
```
Comment (ravirajsubramanian) [inline octopus.go]: Is there no callback for Database calls? How do we handle the error then?

**Review by ravirajsubramanian — COMMENTED** [review]
@thearavind @greenac I have left some comments. Pls, acknowledge if those are necessary.

**Comment by thearavind** [comment]
@greenac @ravirajthedeveloper All the required changes have been done 👍

**Pull request merged**